summingbird-storm/src/main/scala/com/twitter/summingbird/storm/spout/KeyValueSpout.scala

```diff
@@ -4,32 +4,63 @@ import backtype.storm.spout.SpoutOutputCollector
 import backtype.storm.task.TopologyContext
 import backtype.storm.topology.{ IRichSpout, OutputFieldsDeclarer }
 import backtype.storm.tuple.Fields
+import com.twitter.algebird.Semigroup
+import com.twitter.summingbird.online.Externalizer
+import com.twitter.summingbird.online.executor.KeyValueShards
+import com.twitter.summingbird.online.option.SummerBuilder
 import com.twitter.summingbird.storm.Constants._
 import com.twitter.tormenta.spout.SpoutProxy
 import java.util
 import java.util.{ List => JList }
-import com.twitter.summingbird.storm.collector.TransformingOutputCollector
+import scala.collection.mutable.{ MutableList => MList }
+import com.twitter.summingbird.storm.collector.{ AggregatorOutputCollector, TransformingOutputCollector }
 
 /**
  * This is a spout used when the spout is being followed by summer.
- * It uses a TransformingOutputCollector on open.
+ * It uses a AggregatorOutputCollector on open.
  */
 
-class KeyValueSpout(in: IRichSpout) extends SpoutProxy {
+class KeyValueSpout[K, V: Semigroup](val in: IRichSpout, summerBuilder: SummerBuilder, summerShards: KeyValueShards, @transient callOnOpen: (TopologyContext) => Unit) extends SpoutProxy {
+
+  private var adapterCollector: AggregatorOutputCollector[K, V] = _
+  val lockedFn = Externalizer(callOnOpen)
+  var lastDump = System.currentTimeMillis()
```

Comment:
We should depend on a clock abstraction rather than System.currentTimeMillis, otherwise it will be hard to write tests. I'm trying to look for an existing utility like that, we can build one if there isn't one already.